linera-views/src/backends/lru_caching.rs

```diff
@@ -36,19 +36,16 @@ static NUM_CACHE_FAULT: LazyLock<IntCounterVec> =
 static NUM_CACHE_SUCCESS: LazyLock<IntCounterVec> =
     LazyLock::new(|| register_int_counter_vec("num_cache_success", "Number of cache success", &[]));
 
-/// The `LruPrefixCache` stores the data for simple `read_values` queries
-/// It is inspired by the crate `lru-cache`.
+/// Stores the data for simple `read_values` queries.
 ///
-/// We cannot apply this crate directly because the batch operation
-/// need to update the cache. In the case of `DeletePrefix` we have to
-/// handle the keys by prefixes. And so we need to have a `BTreeMap` to
-/// keep track of this.
-
-/// The data structures
+/// This data-structure is inspired by the crate `lru-cache` but was modified to support
+/// range deletions.
 struct LruPrefixCache {
     map: BTreeMap<Vec<u8>, Option<Vec<u8>>>,
     queue: LinkedHashMap<Vec<u8>, (), RandomState>,
     max_cache_size: usize,
+    /// Whether we have exclusive R/W access to the keys under the root key of the store.
+    has_exclusive_access: bool,
```

Comment:
The use of the terminology `has_exclusive_access` is adequate for the terminology.
But the use in the case `!use_exclusive_access` makes sense only in a system where data is written in keys are only written, never modified, never erased. That needs to be said. But yes, it is better than `cache_key_absence` or PR #3496.

The problem of this design is that it forces an API. Conceivably, we could do a `clone_with_root_key` and not have exclusive access to it. Also, we could access to a store and have access to it. Therefore, I think it would make sense to have that in the input parameter to the `LruCachingConfig`.

Reply:
We need to rename `clone_with_root_key`. In fact the types returned by `connect` and `clone_with_root_key` should probably be different.